### PR TITLE
fix: support default export in config files for ESM repos

### PR DESCRIPTION
**Pull request by prescottprue**
:tickets: TICKET-000 

## Changes
* Support default exports within config files (now used in repos which are native ESM):
  * `commitlint.config.js`
  * `lint-staged.config.js`
  * `release.config.js`
* Add import extension support for `.cjs`
* Add nested import support for `vitest.config.js` (importing `vitest/config` triggers import/no-unresolved due to package imports)

## Recordings/Screenshots

image

### CODE REVIEW

File: packages/eslint-config-base/rules/imports.js [hunk 1]
```diff
@@ -142,7 +142,13 @@ module.exports = {
     ...getNoExtraneousDepsOverrides(),
     // This override enables default exports for files that require them.
     {
-      files: ['jest.config.{js,ts,mjs,cjs}', 'vitest.config.{js,ts,mjs,cjs}'],
+      files: [
```
Comment (hershmire) [inline packages/eslint-config-base/rules/imports.js:145]: Looks good to me. While you're at it, can you add `cjs` to our extensions as well? I was about to do another PR for that.

Comment (prescottprue) [reply packages/eslint-config-base/rules/imports.js:145]: Oh yeah good callout - added

**Review by hershmire — APPROVED** [review]
